bundler/helpers/v1/build

```diff
@@ -20,6 +20,6 @@ cd "$install_dir"
 
 # NOTE: Sets `BUNDLED WITH` to match the installed v1 version in Gemfile.lock
 # forcing native helpers to run with the same version
-BUNDLER_VERSION=1 bundle config set --local path ".bundle"
-BUNDLER_VERSION=1 bundle config set --local without "test"
-BUNDLER_VERSION=1 bundle install
+BUNDLER_VERSION=1.17.3 bundle config set --local path ".bundle"
+BUNDLER_VERSION=1.17.3 bundle config set --local without "test"
+BUNDLER_VERSION=1.17.3 bundle install
```

Comment:
I wonder if we should smooth the process of updating bundler in future by having these values in text files we can read in various places?

It might be worth that or a helper script to do bundler updates, but given we have to run things in different bundler contexts that might get tricky.

Reply:
I actually started on a script, but the only approach I can come up with is pretty crude and brittle, along the lines of:

```ruby
uri = URI.parse("https://rubygems.org/api/v1/versions/bundler.json")
response = Net::HTTP.get_response(uri)
body = JSON.parse(response.body)

versions = body.map { |d| Gem::Version.new(d["number"]) }
latest_v2_release = versions.filter { |d| d.canonical_segments.first == 2 }.max
latest_v1_release = versions.filter { |d| d.canonical_segments.first == 1 }.max

dockerfile = File.read("Dockerfile")
dockerfile.gsub!(/\&\& gem install bundler -v 1.+$/, %(&& gem install bundler -v #{latest_v1_release} --no-document \\))
dockerfile.gsub!(/\&\& gem install bundler -v 2.+$/, %(&& gem install bundler -v #{latest_v2_release} --no-document \\))
IO.write("Dockerfile", dockerfile)
```

I'm not sure if such a script is worth maintaining, as we'll also need to update it as our code changes, I'd expect changes to the script would end up taking roughly as much time as going through the files?

There might be a better approach?

Reply:
Nice, I agree that making that maintainable and future proof isn't free compared to a find-and-replace. I don't think it should block this change to solve that either, we might not change this often enough it's worth the metaprogramming 🤔